service/init.go

```diff
@@ -534,3 +541,86 @@ func (amf *AMF) SendNFProfileUpdateToNrf() {
+	if ue != nil && ue.State[models.AccessType__3_GPP_ACCESS].Is(context.Registered) {
+		var nwInitieadDetach bool
+		// Triggers for NwInitiatedDeRegistration
+		// - Only 1 Active SmContext is exist and its slice information matched
+		// - No SmContexts Exists
+		ns := msg.(*protos.NetworkSlice)
+		if len(ue.AllowedNssai[models.AccessType__3_GPP_ACCESS]) == 1 {
+			nwInitieadDetach = true
+		} else {
+
+			//TODO send configuration update to update allowed nssai list
+			//     so pdu session sync up happen in registration procedure, ue should trigger this
+			//gmm_message.SendConfigurationUpdateCommand(ue, models.AccessType__3_GPP_ACCESS, nil)
+
+			//we are doing local cleanup for now, below code will be deprecated when we support
+			// Configuration Update
+			sst, _ := strconv.Atoi(ns.Nssai.Sst)
+			ue.SmContextList.Range(func(key, value interface{}) bool {
+				smContext := value.(*context.SmContext)
+				if smContext.Snssai().Sst == int32(sst) && smContext.Snssai().Sd == ns.Nssai.Sd {
+					logger.CfgLog.Infof("Deleted Slice [sst: %v, sd: %v]matched with smcontext, sending Releasing SMContext Request",
+						smContext.Snssai().Sst, smContext.Snssai().Sd)
+					//send smcontext release request
+					problemDetail, err := consumer.SendReleaseSmContextRequest(ue, smContext, nil, "", nil)
+					if problemDetail != nil {
+						ue.GmmLog.Errorf("Release SmContext Failed Problem[%+v]", problemDetail)
+					} else if err != nil {
+						ue.GmmLog.Errorf("Release SmContext Error[%v]", err.Error())
+					}
+
+				}
+				return true
+			})
+
+			var allowedList []models.AllowedSnssai
+			//update Allowed Nssai List
+			for _, slice := range ue.AllowedNssai[models.AccessType__3_GPP_ACCESS] {
+				if slice.AllowedSnssai.Sst != int32(sst) && slice.AllowedSnssai.Sd != ns.Nssai.Sd {
+					allowedList = append(allowedList, slice)
+				}
+			}
+			ue.AllowedNssai[models.AccessType__3_GPP_ACCESS] = allowedList
+		}
+		// TODO:more than one slice, trigger UEConfigurationUpdate Procedure
+		if nwInitieadDetach {
+			gmm.GmmFSM.SendEvent(ue.State[models.AccessType__3_GPP_ACCESS], gmm.NwInitiatedDeregistrationEvent, fsm.ArgsType{
+				gmm.ArgAmfUe:      ue,
+				gmm.ArgAccessType: models.AccessType__3_GPP_ACCESS,
+			})
+		}
+	} else {
+		logger.CfgLog.Warnf("Ue is not in Registered State")
+		// TODO post Abort event to FSM, handling of Abort event in all States
+	}
+}
+func HandleImsiDeleteFromNetworkSlice(slice *protos.NetworkSlice) {
+	var ue *context.AmfUe
+	var ok bool
+	logger.CfgLog.Infof("[AMF] Handle Subscribers Delete From Network Slice [sst:%v sd:%v]", slice.Nssai.Sst, slice.Nssai.Sd)
+
+	for _, supi := range slice.DeletedImsis {
+		amfSelf := context.AMF_Self()
+		ue, ok = amfSelf.AmfUeFindBySupi("imsi-" + supi)
+		if !ok {
+			logger.CfgLog.Infof("the UE [%v] is not Registered with the 5G-Core", supi)
+			continue
+		}
+		//publish the event to ue channel
+		configMsg := context.ConfigMsg{
+			Supi: supi,
```

Comment:
Possible to include ue-context itself as 2 lookups happening in current flow via amfSelf.AmfUeFindBySupi()
1st called in HandleImsiDeleteFromNetworkSlice
2nd called in  UeConfigHandler()

Reply:
yes .. required in two places, i will optimise in my next commits